mobly/suite_runner.py

```diff
@@ -39,143 +39,143 @@
-                runner.run()
-                ok = runner.results.is_all_pass and ok
-            except signals.TestAbortAll:
-                pass
-            except:
-                logging.exception('Exception when executing %s.',
-                                  config.testbed_name)
-                ok = False
-    if not ok:
-        sys.exit(1)
+  """Executes multiple test classes as a suite.
+
+  This is the default entry point for running a test suite script file
+  directly.
+
+  Args:
+    test_classes: List of python classes containing Mobly tests.
+    argv: A list that is then parsed as cli args. If None, defaults to cli
+      input.
+  """
+  # Parse cli args.
+  parser = argparse.ArgumentParser(description='Mobly Suite Executable.')
+  parser.add_argument(
+      '-c',
+      '--config',
+      type=str,
+      required=True,
+      metavar='<PATH>',
+      help='Path to the test configuration file.')
+  parser.add_argument(
+      '--tests',
+      '--test_case',
+      nargs='+',
+      type=str,
+      metavar='[ClassA[.test_a] ClassB[.test_b] ...]',
+      help='A list of test classes and optional tests to execute.')
+  if not argv:
+    argv = sys.argv[1:]
+  args = parser.parse_args(argv)
+  # Load test config file.
+  test_configs = config_parser.load_test_config_file(args.config)
+
+  # Check the classes that were passed in
+  for test_class in test_classes:
+    if not issubclass(test_class, base_test.BaseTestClass):
+      logging.error(
+          'Test class %s does not extend '
+          'mobly.base_test.BaseTestClass', test_class)
+      sys.exit(1)
+
+  # Find the full list of tests to execute
+  selected_tests = compute_selected_tests(test_classes, args.tests)
+
+  # Execute the suite
+  ok = True
+  for config in test_configs:
+    runner = test_runner.TestRunner(config.log_path, config.testbed_name)
+    with runner.mobly_logger():
+      for (test_class, tests) in selected_tests.items():
+        runner.add_test_class(config, test_class, tests)
+      try:
+        runner.run()
+        ok = runner.results.is_all_pass and ok
+      except signals.TestAbortAll:
+        pass
+      except:
+        logging.exception('Exception when executing %s.', config.testbed_name)
+        ok = False
+  if not ok:
+    sys.exit(1)
 
 
 def compute_selected_tests(test_classes, selected_tests):
-    """Computes tests to run for each class from selector strings.
-
-    This function transforms a list of selector strings (such as FooTest or
-    FooTest.test_method_a) to a dict where keys are test_name classes, and
-    values are lists of selected tests in those classes. None means all tests in
-    that class are selected.
-
-    Args:
-        test_classes: list of strings, names of all the classes that are part
-            of a suite.
-        selected_tests: list of strings, list of tests to execute. If empty,
-            all classes `test_classes` are selected. E.g.
-
-            .. code-block:: python
```

Comment:
this block seems to have been moved to a different place?
are we sure the pyformat is reliable, given all the readthedocs markings we have?